scipy/stats/tests/test_stats.py

```diff
@@ -5596,31 +5596,6 @@ def test_2d_axis1(self):
         desired = array([v, v, v])
         check_equal_gmean(a, desired, axis=1, rtol=1e-14)
 
-    def test_2d_matrix_axis0(self):
```

Comment:
Before merging the masked array and `axis` tuple support, these tests passed, now these tests fail. The reason is that now we convert all array_like objects to regular `ndarrays` before working on them. So before, when the statistic was taken along an axis of the `matrix`, the `matrix` retained its second axis; now the second axis is eliminated when the statistic is taken along it (as is the rule for arrays).

I think that's fine. Although there is precedent for this behavior retains its second axis, there are a few reasons I don't think we need to support it.
1. Special support for `matrix` objects retaining all axes does not appear to be documented anywhere. 
2. Special support for `matrix` objects is not universal. The only other tests in `test_stats.py` that check for this sort of behavior with `matrix` objects are those for `hmean`, which appear to have been written by the same author. `np.mean(np.matrix(a), axis=0)` and `np.var(np.matrix(a), axis=0)` retain their second axis, but `stats.skew(np.matrix(a), axis=0)` and `stats.kurtosis(np.matrix(a), axis=0)` don't!
3. Support for `matrix` objects is declining in general. NumPy [no longer recommends](https://numpy.org/doc/stable/reference/generated/numpy.matrix.html) using the `matrix` class, even for linear algebra. I can't imagine it was ever considered good practice to use it for stats. 

So technically this could be considered a backward-compatibility break, but hopefully you'll agree that it's OK here.

Reply:
+1 from me. I believe we shouldn't be writing new code with `matrix` objects in mind. Off the top of my head the only place where `matrix` is used significantly is `scipy.sparse` and even that seems to be changing.

Reply:
I agree too. It's OK not to support `matrix` now. How about adding a note about it in the docs? Something like "`matrix` objects aren't supported as of 1.9.0" would work.

Reply:
OK, done, but ultimately we're going to want to link to a document that details all this stuff rather than including all the rules in the documentation of every function. Maybe we'll just add something to all the notes like

> For information about support for multi-dimensional array_like objects with ``axis`` and ``nan_policy`` arguments, see here.

Reply:
Added release note information [here](https://github.com/scipy/scipy/wiki/Release-note-entries-for-SciPy-1.9.0). We can fill in a list of all the functions at release time.